### PR TITLE
Protobuf conversion functions for Fog Ledger Router

### PR DESCRIPTION
**Pull request by awygle**
Pull in protobuf object conversion functions for Fog Ledger Router

### Motivation

As part of the Fog Ledger Router project, we need functions to convert from protobuf objects to various Rust objects. This pulls in the changes made to conversions.rs for upstreaming.

### Future Work

The next PR in this series will be the stubs of the APIs added to the Fog Ledger enclave to support the Fog Ledger Router.

### CODE REVIEW

**Comment by awygle** [comment]
Current dependencies on/for this PR:
* feature/fog-ledger-router
  * **PR #2889** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2889" target="_blank">Graphite</a> 
    * **PR #2891** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2891" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2891?utm_source=stack-comment).

File: fog/api/src/conversions.rs [hunk 1]
```diff
@@ -30,6 +32,25 @@ impl From<Vec<attest::NonceMessage>> for MultiViewStoreQueryRequest {
...
+    fn from(enclave_messages: Vec<EnclaveMessage<NonceSession>>) -> MultiKeyImageStoreRequest {
+        enclave_messages
+            .into_iter()
+            .map(|enclave_message| enclave_message.into())
+            .collect::<Vec<attest::NonceMessage>>()
+            .into()
+    }
+}
+
+impl From<Vec<attest::NonceMessage>> for MultiKeyImageStoreRequest {
```
Comment (nick-mobilecoin) [inline fog/api/src/conversions.rs:45]: :question: I'm not opposed to this implementation as I'm generally pro functionalizing things.
Just curious if there is another use case for `Vec<attest::NonceMessage>` -> `MultiKeyImageStoreRequest` outside of the `Vec<EnclaveMessage<NonceSession>>` -> `MultiKeyImageStoreRequest`

**Review by nick-mobilecoin — APPROVED** [review]

File: fog/api/src/conversions.rs (hunk 1 above)
Comment (NotGyro) [reply fog/api/src/conversions.rs:45]: A lot of Rust code (I can't be any less vague than this, I just see it very often) is genericized such that you can pass any type which implements `From<T>` to certain functions, to provide some extra convenience. `From` is reflexive, so you can pass `T` or anything that can be converted to `T` to a function which takes `From<T>` like that. 

This is just ergonomics, of course, and having a function that does this explicitly might be clearer / more readable than having the type system do it for you implicitly. I wouldn't really mind either way.

**Review by NotGyro — APPROVED** [review]
Looks good to me - although I wrote the first draft of this code, if I'm remembering correctly, so take my review with a grain of salt.

File: fog/api/src/conversions.rs (hunk 1 above)
Comment (awygle) [reply fog/api/src/conversions.rs:45]: The conversions between AKE messages and domain-specific messages are pretty hairy in general, IMO. I believe that I copied an existing pattern for this.

We still `cargo check` without the `From<Vec<attest::NonceMessage>>` impl, so lifting the actual `MultiKeyImageStoreRequest` stuff into the other `From` block would work fine. But I think it's fine to leave this as-is.

Comment (nick-mobilecoin) [reply fog/api/src/conversions.rs:45]: sounds good I don't really have a strong preference one way or another

**Pull request merged**